superset/assets/visualizations/nvd3_vis.js

```diff
@@ -665,16 +676,29 @@ function nvd3Vis(slice, payload) {
             const tip = tipFactory(e);
 
             const records = (slice.annotationData[e.name].records || []).map((r) => {
-              const timeColumn = new Date(r[e.timeColumn]);
-              const intervalEndColumn = new Date(r[e.intervalEndColumn]);
+              const timeValue = new Date(r[e.timeColumn]);
+              const intervalEndValue = new Date(r[e.intervalEndColumn]);
               return {
                 ...r,
-                [e.timeColumn]: timeColumn,
-                [e.intervalEndColumn]: intervalEndColumn,
+                [e.timeColumn]: timeValue,
+                [e.intervalEndColumn]: intervalEndValue,
               };
-            }).filter(r => !Number.isNaN(r[e.timeColumn].getMilliseconds()) &&
-              !Number.isNaN(r[e.intervalEndColumn].getMilliseconds()));
+            }).filter((r) => {
+              const isValid = (
+                !Number.isNaN(r[e.timeColumn].getMilliseconds()) &&
+                !Number.isNaN(r[e.intervalEndColumn].getMilliseconds())
+              );
+              if (isValid) {
+                xMin = xMin < r[e.timeColumn] ? xMin : r[e.timeColumn];
```

Comment:
The code currently does not really check that `timeColumn` < `intervalEndColumn` and will flip them around when necessary. Not sure if we need to handle that though.